addons/l10n_de/models/ir_attachment.py

```diff
@@ -49,16 +51,19 @@ def unlink(self):
             attachment.res_model == 'account.move'
             and attachment.res_id
             and attachment.res_field in ('invoice_pdf_report_file', 'ubl_cii_xml_file')
+            and attachment.company_id.country_id.code == 'DE'
```

Comment:
Add the company in the list of fields that can't be changed in `write`.
This will still add a bypass if we change the country of the company before the unlink...